app/domain/authentication/o_auth/fetch_provider_keys.rb

```diff
@@ -29,8 +29,14 @@ def discovered_provider
       end
 
       def fetch_provider_keys
+        # We have to invoke the custom discovery service to be able to fetch the jwks
+        # since it also requires the CA to invoke the jwks URI
         jwks = {
-          keys: @discovered_provider.jwks
```

Comment:
@john-odonnell This is still a WIP, but I'm curious if you have thoughts about this: turns out that `.jwks` makes a separate call to the provider's JWKS uri so the cert needs to be available for that as well.

As a workaround I added an optional `jwks` flag to the custom `Client.discover` which updates the lambda to include the `.jwks` call accordingly. Feels a bit hacky but better than the alternative of adding yet another wrapper imo. 

At the same time I'm wondering if is worth refactoring the wrapper functionality into a single method that can perform one of three operations: `[:callback, :discover, :jwks]`. Would require better validation of the inputs but may look cleaner than the duplicated code we have currently

Reply:
Otherwise, you're right - feels hacky, but I think this works for now. Makes me think that ditching the underlying OIDC client gem might make all this a bit cleaner, though.